### DISCUSSION TITLE
Proposal for Enhancing Upstream Shell Completion Feature Visibility

### DISCUSSION THREAD

**Discussion by yoshida-ryuhei** [fish-shell/fish-shell#10122] [Ideas]
Hi there,

I want to express my gratitude for the fantastic completion feature in the Fish shell.

I've been thinking about improving the visibility of this feature by creating almost empty completion files for various tools and proposing them for inclusion upstream. In some completion files, like [share/completions/docker.fish](https://github.com/fish-shell/fish-shell/blob/master/share/completions/docker.fish), the upstream completions feature is mentioned. I believe making pull requests with minimal completion files could effectively showcase that the CLI tool supports completions natively.

I've noticed that finding instructions for installing shell completions from upstream packages can be challenging at times. Therefore, including a brief introduction in the completion files could be incredibly helpful. This way, users would immediately know that the CLI tool offers built-in completions.

Please let me know your thoughts on this proposal, and if you think it's a good idea, I'd be happy to start creating and submitting those pull requests.

**Comment by faho** [comment]
> In some completion files, like [share/completions/docker.fish](https://github.com/fish-shell/fish-shell/blob/master/share/completions/docker.fish), the upstream completions feature is mentioned

Most people won't look at that file. The reason that specific file has the contents it has is because people kept trying to add them otherwise.

It isn't a good way to tell people that a thing can be completed by running some magical incantation, and it isn't typically helpful to include that because most files don't have the problem that the docker completions had.

> I've noticed that finding instructions for installing shell completions from upstream packages can be challenging at times

The best solution for that is to get your completion scripts to be *installed* rather than asking people to run `fooctl completion --fish | source` - but if we have to do that, [we will](https://github.com/fish-shell/fish-shell/blob/ccc8308d41d7b578fe3102e942049b8e7d1a5228/share/completions/doctl.fish).

Your best bet is therefore to go to projects that have awkward "install shell completion" instructions, and get them to improve that step. Go to your distro packagers and have them pre-generate the files.

**Reply by yoshida-ryuhei** [reply]
Thank you for the quick reply. 

I understand your idea very well and I agree with it. 
I will ask the upstream first, and if they give no response, I may write pull requests like [doctl.fish](https://github.com/fish-shell/fish-shell/blob/ccc8308d41d7b578fe3102e942049b8e7d1a5228/share/completions/doctl.fish) you mentioned.
